pkgs/applications/misc/phoc/default.nix

```diff
@@ -0,0 +1,68 @@
+{ stdenv
+, fetchFromGitLab
+, meson
+, ninja
+, pkgconfig
+, wrapGAppsHook
+, libinput
+, gnome3
+, glib
+, gtk3
+, wayland
+, dbus
+, cmake
+, libdrm
+, libxkbcommon
+, wlroots
+}:
+
+stdenv.mkDerivation rec {
+  pname = "phoc";
+  version = "0.1.7";
+
+  src = fetchFromGitLab {
+    domain = "source.puri.sm";
+    owner = "Librem5";
+    repo = pname;
+    rev = "3d68b091d0075bd549f6fbb84d6c152f01a2354e";
+    sha256 = "1007kxap87h1bi79sz45lfb478d93pq0676cr5rnhbanljm1n28n";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    meson
+    ninja
+    pkgconfig
+    wrapGAppsHook
+  ];
+
+  buildInputs = [
+    libdrm.dev
+    libxkbcommon
+    libinput
+    glib
+    gtk3
+    gnome3.gnome-desktop
+    gnome3.mutter
```

Comment:
Does it  really depend on two different compositor libraries (mutter & wlroots)?

Reply:
Without `mutter` fails with this error:
```
Jun 01 13:25:34 nixos .gnome-shell-wr[956]: ../clutter/clutter/clutter-actor.c:10548: The clutter_actor_set_allocation() function can only be called from within the implementation of the ClutterActor::allocate() virtual function.
Jun 01 13:25:35 nixos gdm-password][1243]: gkr-pam: unable to locate daemon control file
Jun 01 13:25:35 nixos gdm-password][1243]: gkr-pam: stashed password to try later in open session
Jun 01 13:25:35 nixos gdm-password][1243]: pam_unix(gdm-password:session): session opened for user jordi by (uid=0)
Jun 01 13:25:35 nixos systemd[1]: Started Session 7 of user jordi.
Jun 01 13:25:35 nixos systemd-logind[673]: New session 7 of user jordi.
Jun 01 13:25:35 nixos gnome-keyring-daemon[1234]: module_instances: assertion 'funcs != NULL && "instances"' failed
Jun 01 13:25:35 nixos gnome-keyring-daemon[1234]: module_instances: assertion 'funcs != NULL && "instances"' failed
Jun 01 13:25:35 nixos gnome-keyring-daemon[1234]: lookup_login_keyring: assertion 'GCK_IS_SESSION (session)' failed
Jun 01 13:25:35 nixos gnome-keyring-daemon[1234]: create_credential: assertion 'GCK_IS_SESSION (session)' failed
Jun 01 13:25:35 nixos gnome-keyring-daemon[1234]: egg_error_message: assertion 'error' failed
Jun 01 13:25:35 nixos gnome-keyring-daemon[1234]: couldn't create login credential: (unknown)
Jun 01 13:25:35 nixos gdm-password][1243]: gkr-pam: the password for the login keyring was invalid.
Jun 01 13:25:35 nixos kernel: rfkill: input handler enabled
Jun 01 13:25:35 nixos /nix/store/k4rhsv9ixqqvgvcn0zigsxnr042x6m1x-gdm-3.34.1/libexec/gdm-wayland-session[1268]: No schemas installed
Jun 01 13:25:36 nixos /nix/store/k4rhsv9ixqqvgvcn0zigsxnr042x6m1x-gdm-3.34.1/libexec/gdm-wayland-session[1250]: pci id for fd 9: 1234:1111, driver (null)
Jun 01 13:25:36 nixos .phoc-wrapped[1250]: Settings schema 'org.gnome.mutter.keybindings' is not installed
Jun 01 13:25:36 nixos kernel: traps: .phoc-wrapped[1250] trap int3 ip:7efc7ce11405 sp:7fff6c988440 error:0 in libglib-2.0.so.0.6400.1[7efc7cdd7000+81000]
Jun 01 13:25:36 nixos systemd[1]: Started Process Core Dump (PID 1273/UID 0).
Jun 01 13:25:36 nixos gdm-password][1243]: pam_unix(gdm-password:session): session closed for user jordi
Jun 01 13:25:36 nixos .gnome-shell-wr[956]: Failed to set hardware cursor (drmModeSetCursor failed: No such device or address), using OpenGL from now on
Jun 01 13:25:36 nixos gdm[837]: GdmDisplay: Session never registered, failing
Jun 01 13:25:36 nixos kernel: rfkill: input handler disabled
Jun 01 13:25:37 nixos systemd[1]: session-7.scope: Succeeded.
Jun 01 13:25:37 nixos systemd-logind[673]: Session 7 logged out. Waiting for processes to exit.
Jun 01 13:25:37 nixos systemd-logind[673]: Removed session 7.
Jun 01 13:25:37 nixos systemd-coredump[1274]: Process 1250 (.phoc-wrapped) of user 1000 dumped core.
Jun 01 13:25:37 nixos systemd[1]: systemd-coredump@2-1273-0.service: Succeeded.
Jun 01 13:25:37 nixos gnome-shell[1282]: The XKEYBOARD keymap compiler (xkbcomp) reports:
Jun 01 13:25:37 nixos gnome-shell[1282]: > Warning:          Unsupported maximum keycode 569, clipping.
Jun 01 13:25:37 nixos gnome-shell[1282]: >                   X11 cannot support keycodes above 255.
Jun 01 13:25:37 nixos gnome-shell[1282]: > Warning:          Unsupported high keycode 372 for name <I372> ignored
Jun 01 13:25:37 nixos gnome-shell[1282]: >                   X11 cannot support keycodes above 255.
Jun 01 13:25:37 nixos gnome-shell[1282]: >                   This warning only shows for the first high keycode.
Jun 01 13:25:37 nixos gnome-shell[1282]: Errors from xkbcomp are not fatal to the X server
```

Reply:
done